pinot-core/src/main/java/com/linkedin/pinot/core/minion/BackfillDateTimeColumn.java

```diff
@@ -240,7 +243,10 @@ public GenericRow next(GenericRow row) {
      */
     private Object convertTimeFieldToDateTimeFieldSpec(Object timeColumnValue) {
 
-      TimeGranularitySpec timeGranularitySpec = _timeFieldSpec.getIncomingGranularitySpec();
+      TimeGranularitySpec timeGranularitySpec = _timeFieldSpec.getOutgoingGranularitySpec();
+      if (timeGranularitySpec == null) {
```

Comment:
getOutgoingGranularitySpec already does this null check and returns incoming granularity spec

Reply:
Ah nice, removed my redundant check